bundles/org.openhab.binding.noolite/src/main/java/org/openhab/binding/noolite/internal/NooliteMTRF64Adapter.java

```
@@ -0,0 +1,114 @@
+
+            in = new DataInputStream(serial.getInputStream());
+            output = serial.getOutputStream();
+            serial.addEventListener(new SerialPortEventListener() {
+                @Override
+                public void serialEvent(SerialPortEvent event) {
+                    try {
+                        byte[] data = new byte[17];
+                        if (in.read(data) > 0) {
+                            logger.debug("Received data: {}", DatatypeConverter.printHexBinary(data));
+                            short count = 0;
+                            byte sum = 0;
+                            for (int i = 0; i <= 14; i++) {
+                                count += (data[i] & 0xFF);
+                            }
+                            sum = (byte) (count & 0xFF);
+
+                            logger.debug("sum is {} CRC must be {} receive {}", count, sum, data[15]);
+
+                            if (((data[0] & 0xFF) == 0b10101101) && ((data[16] & 0xFF) == 0b10101110)) {
+                                logger.debug("sum is {} CRC must be {} receive {}", count, sum, data[15]);
+                                if (sum == data[15]) {
+                                    logger.debug("CRC is OK");
+
+                                    logger.debug("Updating values...");
+                                    NooliteMTRF64BridgeHandler.updateValues(data);
+                                } else {
+                                    logger.debug("CRC is WRONG");
+                                }
+                            } else {
+                                logger.debug("Start/stop bits is wrong");
+                            }
+                        }
+                    } catch (IOException ex) {
+                        logger.debug("Error reading from serial port!", ex);
+                    }
+                }
+            });
+            serial.notifyOnDataAvailable(true);
+        } catch (Exception e) {
```

Can this be a specific exception? And always log exceptions. You may never know if someone gets this error and if nothing is logged. It will just silently fail.